src/components/organisms/Page.js

```diff
@@ -25,6 +28,12 @@ export function Page(props) {
     }
   };
 
+  useEffect(() => {
+    if (keycloak.authenticated && Object.keys(userProfileData).length === 0) {
+      dispatch(getUserData(keycloak));
+    }
+  }, [keycloak, dispatch, userProfileData]);
+
```

Comment:
Could this not be put into a selector?

Reply:
Hmm hadn't considered that. Do you have a rough example you could give @var-kyle ? Do you mean move the condition into a selector and just do if (selector)?